source/whats-new.txt

```diff
@@ -12,13 +12,38 @@ What's New
 
 Learn what's new in:
 
+* :ref:`Version 4.6 <version-4.6>`
 * :ref:`Version 4.5 <version-4.5>`
 * :ref:`Version 4.4 <version-4.4>`
 * :ref:`Version 4.3 <version-4.3>`
 * :ref:`Version 4.2 <version-4.2>`
 * :ref:`Version 4.1 <version-4.1>`
 * :ref:`Version 4.0 <version-4.0>`
 
+
+.. _version-4.6:
+
+What's New in 4.6
+-----------------
+
+New features of the 4.6 Java driver release include:
+
+- The buffer pool is now shared across all ``MongoClient`` instances. The
+  buffer pool prunes buffers that have been idle for one minute.
+- Added an API to specify suppliers for AWS KMS credentials for
+  Client-Side Field Level Encryption (CSFLE). See the
+  `AutoEncryptionSettings.Builder.kmsProviderPropertySuppliers() <https://mongodb.github.io/mongo-java-driver/4.6/apidocs/mongodb-driver-core/com/mongodb/AutoEncryptionSettings.Builder.html#kmsProviderPropertySuppliers(java.util.Map)>`__
+  and the `ClientEncryptionSettings.Builder.kmsProviderPropertySuppliers() <https://mongodb.github.io/mongo-java-driver/4.6/apidocs/mongodb-driver-core/com/mongodb/ClientEncryptionSettings.html#getKmsProviderPropertySuppliers()>`__
+  API documentation for more information.
+- Added compatibility to ``org.mongodb:mongodb-crypt``, a dependency for
+  CSFLE, for Apple M1 ARM-based hardware.
+- Added a service provider interface (SPI) to allow the driver to specify
+  a custom DNS client. See the `DnsClient <https://mongodb.github.io/mongo-java-driver/4.6/apidocs/mongodb-driver-core/com/mongodb/spi/dns/DnsClient.html>`__
+  and the `DnsClientProvider <https://mongodb.github.io/mongo-java-driver/4.6/apidocs/mongodb-driver-core/com/mongodb/spi/dns/DnsClientProvider.html>`__
+  interface API documentation for more information.
+- Added the `` org.mongodb:bson-record-codec`` library for encoding and
```

Comment:
I'm not sure this is the best way to document this, for a couple of reasons

1. RecordCodec is not a public class
2. RecordCodecProvider is including in the default codec registry, so to make use of it you don't have to use RecordCodecProvider directly at all.

We should say something like:

> The driver supports encoding and decoding Java records (link) to and from BSON documents.  This support is enabled by default.

And perhaps link to API docs for com.mongodb.MongoClientSettings#getDefaultCodecRegistry

Reply:
@jyemin From browsing Maven, I wasn't able to find which current JAR lists bson-record-codec as a dependency, so if that's the case, don't they need to add it for default support?

Reply:
mongodb-driver-core does: https://repo1.maven.org/maven2/org/mongodb/mongodb-driver-core/4.6.0/mongodb-driver-core-4.6.0.pom